libdnf/module/module_item_container.cpp

```diff
@@ -70,4 +71,37 @@ BaseWeakPtr ModuleItemContainer::get_base() const {
 }
 
 
+InvalidModuleState::InvalidModuleState(const std::string & state)
+    : libdnf::Error(M_("Invalid module state: {}"), state) {}
+
+
+std::string module_state_to_string(ModuleState state) {
+    switch (state) {
+        case ModuleState::AVAILABLE:
+            return "Available";
+        case ModuleState::DEFAULT:
+            return "Default";
+        case ModuleState::ENABLED:
+            return "Enabled";
+        case ModuleState::DISABLED:
+            return "Disabled";
+    }
+    return "";
+}
+
+
+ModuleState module_state_from_string(const std::string & state) {
+    if (state == "Available") {
+        return ModuleState::AVAILABLE;
+    } else if (state == "Default") {
+        return ModuleState::DEFAULT;
+    } else if (state == "Enabled") {
+        return ModuleState::ENABLED;
+    } else if (state == "Disabled") {
+        return ModuleState::DISABLED;
+    }
+
+    throw InvalidModuleState(state);
+}
```

Comment:
In DNF 4, there was empty string for both `AVAILABLE` and `DEFAULT` states, because it doesn't make sense to distinguish between these two states in the config (it depends on the loaded metadata and can change over time). And the strings that are used e.g. in `module list` or `module info` when printing states of modules are different anyway, so these `module_state_to_string` and `module_state_from_string` are only used for the `system::ModuleState`. I think it would be better to have one string for both of these states and interpret it as `AVAILABLE`.

Reply:
I guess that the information is even stored internally on different places.